Client/src/Components/Admin/AdminReviewView.tsx

```diff
@@ -0,0 +1,113 @@
+        await flagDataset(query)
+    }
+
+    const handleApproveDataset = async (datasetId: IApprovedDatasetModel) => {
+        const query: IFlaggedDatasetQuery = { datasetId: dataset.id, additionalComments: comment }
+        await adminApprovedDataset(query)
+        setUpdate(update + 1)
+        handleDatasetChange(null)
+    }
+
+    const handleDatasetChange = (newDataset: IApprovedDatasetModel) => {
+        setDataset(newDataset)
+    }
+
+    const handleCommentChange = (event) => {
+        setComment(event.target.value)
+    }
+
+    const handleFlaggedCommentChange = (event) => {
+        setFlaggedComment(event.target.value)
+    }
+    return (
+
+        <Grid justify="flex-start" container spacing={4}>
+            <Grid xs={3}>
+                <Box p={4} pt={7}>
+                    <Typography variant='h5'>
+                        Datasets to be reviewed
+                    </Typography>
+                    <br></br>
+                    <AdminReviewList
+                        datasets={datasetState}
+                        onChange={handleDatasetChange}
+                        update={update}
+                    />
+                    <br></br>
+                    <Button id="toggle-edit" onClick={handleEditDataset} color="primary" variant="contained">Edit</Button>&nbsp;&nbsp;&nbsp;
+                <Button id="toggle-edit" color="primary" onClick={handleFlagDataset} variant="contained">Flag</Button>&nbsp;&nbsp;&nbsp;
+                <Button id="toggle-edit" color="primary" onClick={handleDeleteDataset} variant="contained">Delete</Button>
+                </Box>
+                <Box p={5}>
+                    <TextField
+                        id="standard-multiline-flexible"
+                        label="Add a Comment"
+                        multiline
+                        fullWidth={true}
+                        rowsMax={4}
+                        value={comment}
+                        onChange={handleCommentChange}
+                    />
+                </Box>
+                <Box p={5}>
+                    <TextField
+                        id="standard-multiline-flexible"
+                        label="Add a Flagged Comment"
+                        multiline
+                        fullWidth={true}
+                        rowsMax={4}
+                        value={flaggedComment}
+                        onChange={handleFlaggedCommentChange}
+                    />
+                </Box>
+            </Grid>
+
+            <Grid xs={8}>
+                <Grid container spacing={3}>
+                    <Grid xs={12}>
+                        {dataset &&
+                            <DatasetUploadForm
+                                onSubmit={handleApproveDataset}
+                                initialDataset={dataset}
+                                editable={editable}
+                                buttonName="Approve Dataset"
```

Comment:
We might want to move this to the top of the dataset instead of the bottom, it is not very obvious that it was there.

Reply:
That'll probably be dealt with when the Dataset Upload Form is refactored. Since currently we're reusing the button that was used to submit a new dataset.

Reply:
Makes sense